### PR TITLE
add some new features

### PR DESCRIPTION
**Pull request by pyltsin**
Hello, I add some new features:

- add work with "Search everywhere"
- add work with Kotlin
- improve support of path variable
- add support feign client
- add support environment variable
- change search logic
- change some deprecated method

I am new to contribution and may be my request is bad. I can improve/delete some parts and other, what you say.
I'm sorry  for second mr, I improved my code style

### CODE REVIEW

**Comment by viartemev** [comment]
@pyltsin thanks a lot for your contribution. I will check it in the near future.

**Comment by viartemev** [comment]
isssue: https://github.com/viartemev/requestmapper/issues/19

**Comment by viartemev** [comment]
@pyltsin there is a problem with resolving path mappings for Kotlin files:
```kotlin
const val SCHEDULE_URL = "/api/v1/schedules"

@PostMapping(SCHEDULE_URL)
fun saveSchedule() {
  //some logic is here
}
```

File: build.gradle [hunk 1]
```diff
@@ -79,7 +79,8 @@ test {
 intellij {
     pluginName "Request mapper"
     type 'IC'
-    version '2019.1'
+    version = '2020.1.1'
+    plugins = ["java", "org.jetbrains.kotlin"]
```
Comment (viartemev) [inline build.gradle:83]: What's the difference with `"org.jetbrains.kotlin:$kotlin_version-release-IJ2020.1-1"` ?

File: src/main/kotlin/com/viartemev/requestmapper/RequestMappingGoToContributor.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.viartemev.requestmapper
+
+import com.intellij.ide.actions.searcheverywhere.AbstractGotoSEContributor
+import com.intellij.ide.actions.searcheverywhere.SearchEverywhereContributor
+import com.intellij.ide.actions.searcheverywhere.SearchEverywhereContributorFactory
+import com.intellij.ide.util.gotoByName.FilteringGotoByModel
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+
+class RequestMappingGoToContributor(event: AnActionEvent) : AbstractGotoSEContributor(event) {
```
[the remark this answers is not in the document]
Comment (pyltsin) [reply src/main/kotlin/com/viartemev/requestmapper/RequestMappingGoToContributor.kt:10]: I'm sorry. My mistake -  I change minimum version to 201.7000

File: src/main/kotlin/com/viartemev/requestmapper/annotations/spring/SpringMappingAnnotation.kt [hunk 3]
```diff
@@ -95,11 +99,13 @@ abstract class SpringMappingAnnotation(
     }
 
     companion object {
+        private const val URL = "url"
         private const val VALUE = "value"
         private const val PATH = "path"
         private const val NAME = "name"
         private const val PARAMS = "params"
         private const val SPRING_REQUEST_MAPPING_CLASS = "org.springframework.web.bind.annotation.RequestMapping"
+        private val SPRING_FEIGN_REQUEST_MAPPING_CLASSES = listOf("org.springframework.cloud.openfeign.FeignClient", "org.springframework.cloud.netflix.feign.FeignClient")
```
[the remark this answers is not in the document]
Comment (pyltsin) [reply src/main/kotlin/com/viartemev/requestmapper/annotations/spring/SpringMappingAnnotation.kt]: delete  feign. I try doing it later.

File: build.gradle (hunk 1 above)
Comment (pyltsin) [reply build.gradle:83]: I think they are equivalent. because Idea shows me that I use 1.3.72-release-IJ2020.1-1 org.jetbrains.kotlin

**Comment by pyltsin** [comment]
> @pyltsin there is a problem with resolving path mappings for Kotlin files:
> 
> ```kotlin
> const val SCHEDULE_URL = "/api/v1/schedules"
> 
> @PostMapping(SCHEDULE_URL)
> fun saveSchedule() {
>   //some logic is here
> }
> ```
> 
> You need add handling of `kt` specific annotations in `BasePsiAnnotationValueVisitor`

Thank you for this case.  I'm not sure (I don't have all examples), but may be it will work - I change extract logic

**Comment by viartemev** [comment]
@pyltsin thanks for the PR. I will release those features in release 0.2 on Monday.
Would you like to share with me your first name and last name? I would like to add you to a contributor list in README.md
I will also be grateful if you make a PR for `Feign` support.

**Comment by pyltsin** [comment]
Thank you for the plugin. My name is Pyltsin Mikhail. If you add to the list, I will be happy. 
About Feign - I will try to find time for it, but it will in June-July.
Thank you again!

**Comment by viartemev** [comment]
@pyltsin I'm not sure that the plugin will be released today, I missed one important thing. In current implementation `kotlin` plugin is necessary for this plugin. It's not correct because the plugin should work without `kotlin` plugin.

**Comment by pyltsin** [comment]
I understand you. Tonight I will search a  solution for it.

File: src/main/kotlin/com/viartemev/requestmapper/annotations/spring/SpringMappingAnnotation.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (harryeti) [reply src/main/kotlin/com/viartemev/requestmapper/annotations/spring/SpringMappingAnnotation.kt]: Or can I do some local customization for `Feign` annotations?

[the remark this answers is not in the document]
Comment (pyltsin) [reply src/main/kotlin/com/viartemev/requestmapper/annotations/spring/SpringMappingAnnotation.kt]: I think, you can use this commit. Sorry, I don't have free time to add this yet.

[the remark this answers is not in the document]
Comment (harryeti) [reply src/main/kotlin/com/viartemev/requestmapper/annotations/spring/SpringMappingAnnotation.kt]: That's ok, I'll try this. Many thanks.

**Pull request merged**